"panic: runtime error: index out of range" when attempting to use dcount in vector runtime

## tl;dr

While filing #5708, I noticed that query panics when executed in vector runtime.

```
$ SUPER_VAM=1 super -c "SELECT dcount(client_ip) FROM 'bench2.parquet'"
panic: runtime error: index out of range [2] with length 2
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:26 +0x5e
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull.func1()
	/Users/phil/work/super/runtime/sam/op/catcher.go:25 +0x3d
panic({0x41bb420?, 0xc000418078?})
	/usr/local/opt/go/libexec/src/runtime/panic.go:792 +0x132
github.com/brimdata/super/vector.(*Bytes).Value(...)
	/Users/phil/work/super/vector/bytes.go:45
github.com/brimdata/super/vector.BytesValue({0x42c0438?, 0xc0067fc9c0?}, 0x1?)
	/Users/phil/work/super/vector/bytes.go:51 +0x3b2
github.com/brimdata/super/runtime/vam/expr/agg.(*dcount).ConsumeAsPartial(0xc00161c120, {0x42c0438, 0xc0067fc9c0})
	/Users/phil/work/super/runtime/vam/expr/agg/dcount.go:47 +0x8c
github.com/brimdata/super/runtime/vam/op/summarize.(*superTable).update(0xc018c68410, {0xc012cc36d0, 0x0, 0x1}, {0xc012cc36d0, 0x1, 0xc000fa14f0?})
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:67 +0x506
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).consume(0xc0003eb1e0, {0xc012cc36d0, 0x0, 0x1}, {0xc012cc36d0, 0x1, 0x1})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:102 +0x27e
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull.func1({0xc012cc36d0, 0x1, 0xc000fa1608?})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:83 +0x75
github.com/brimdata/super/vector.Apply(0x0?, 0x42c0258?, {0xc012cc36d0?, 0x30?, 0x407fb80?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc0003eb1e0, 0xb3?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:82 +0x13b
github.com/brimdata/super/runtime/vam/op.(*Yield).Pull(0xc0014df3b0, 0x0)
	/Users/phil/work/super/runtime/vam/op/yield.go:27 +0x44
github.com/brimdata/super/runtime/vam.(*Materializer).Pull(0xc0005ac0e0?, 0x0?)
	/Users/phil/work/super/runtime/vam/materialize.go:26 +0x27
github.com/brimdata/super/runtime/sam/op.(*Single).Pull(0xc0014df410, 0x5?)
	/Users/phil/work/super/runtime/sam/op/mux.go:120 +0x33
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull(0x4744005b20522595?, 0x59?)
	/Users/phil/work/super/runtime/sam/op/catcher.go:28 +0x5c
github.com/brimdata/super/runtime/exec.(*Query).Pull(0x234d0df?, 0x0?)
	/Users/phil/work/super/runtime/exec/query.go:49 +0x3c
github.com/brimdata/super/zbuf.CopyMux(0xc000fa1d68, {0x42ac600, 0xc0014df440})
	/Users/phil/work/super/zbuf/mux.go:40 +0x38
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0003ae600, {0xc00003c710, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:167 +0x996
github.com/brimdata/super/pkg/charm.path.run({0xc000080c08, 0x1, 0x1}, {0xc00003c710, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x53e9e60, {0xc00003c700, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:71 +0x105
main.main()
	/Users/phil/work/super/cmd/super/main.go:40 +0x5b
```

## Details

Repro is with super commit baeaf90e.

It looks like the problem is not format-specific, as I get a similar panic when querying against Parquet instead of CSUP as I showed above.

```
$ super -version
Version: v1.18.0-322-gbaeaf90e

$ SUPER_VAM=1 super -c "SELECT dcount(client_ip) FROM 'bench2.parquet'"
panic: runtime error: index out of range [2] with length 2
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/opt/go/libexec/src/runtime/debug/stack.go:26 +0x5e
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull.func1()
	/Users/phil/work/super/runtime/sam/op/catcher.go:25 +0x3d
panic({0x41bb420?, 0xc000418078?})
	/usr/local/opt/go/libexec/src/runtime/panic.go:792 +0x132
github.com/brimdata/super/vector.(*Bytes).Value(...)
	/Users/phil/work/super/vector/bytes.go:45
github.com/brimdata/super/vector.BytesValue({0x42c0438?, 0xc0067fc9c0?}, 0x1?)
	/Users/phil/work/super/vector/bytes.go:51 +0x3b2
github.com/brimdata/super/runtime/vam/expr/agg.(*dcount).ConsumeAsPartial(0xc00161c120, {0x42c0438, 0xc0067fc9c0})
	/Users/phil/work/super/runtime/vam/expr/agg/dcount.go:47 +0x8c
github.com/brimdata/super/runtime/vam/op/summarize.(*superTable).update(0xc018c68410, {0xc012cc36d0, 0x0, 0x1}, {0xc012cc36d0, 0x1, 0xc000fa14f0?})
	/Users/phil/work/super/runtime/vam/op/summarize/agg.go:67 +0x506
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).consume(0xc0003eb1e0, {0xc012cc36d0, 0x0, 0x1}, {0xc012cc36d0, 0x1, 0x1})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:102 +0x27e
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull.func1({0xc012cc36d0, 0x1, 0xc000fa1608?})
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:83 +0x75
github.com/brimdata/super/vector.Apply(0x0?, 0x42c0258?, {0xc012cc36d0?, 0x30?, 0x407fb80?})
	/Users/phil/work/super/vector/apply.go:17 +0xa3
github.com/brimdata/super/runtime/vam/op/summarize.(*Summarize).Pull(0xc0003eb1e0, 0xb3?)
	/Users/phil/work/super/runtime/vam/op/summarize/summarize.go:82 +0x13b
github.com/brimdata/super/runtime/vam/op.(*Yield).Pull(0xc0014df3b0, 0x0)
	/Users/phil/work/super/runtime/vam/op/yield.go:27 +0x44
github.com/brimdata/super/runtime/vam.(*Materializer).Pull(0xc0005ac0e0?, 0x0?)
	/Users/phil/work/super/runtime/vam/materialize.go:26 +0x27
github.com/brimdata/super/runtime/sam/op.(*Single).Pull(0xc0014df410, 0x5?)
	/Users/phil/work/super/runtime/sam/op/mux.go:120 +0x33
github.com/brimdata/super/runtime/sam/op.(*Catcher).Pull(0x4744005b20522595?, 0x59?)
	/Users/phil/work/super/runtime/sam/op/catcher.go:28 +0x5c
github.com/brimdata/super/runtime/exec.(*Query).Pull(0x234d0df?, 0x0?)
	/Users/phil/work/super/runtime/exec/query.go:49 +0x3c
github.com/brimdata/super/zbuf.CopyMux(0xc000fa1d68, {0x42ac600, 0xc0014df440})
	/Users/phil/work/super/zbuf/mux.go:40 +0x38
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0003ae600, {0xc00003c710, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:167 +0x996
github.com/brimdata/super/pkg/charm.path.run({0xc000080c08, 0x1, 0x1}, {0xc00003c710, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x53e9e60, {0xc00003c700, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:71 +0x105
main.main()
	/Users/phil/work/super/cmd/super/main.go:40 +0x5b
```

--- a/runtime/vam/expr/agg/dcount.go
+++ b/runtime/vam/expr/agg/dcount.go
@@ -44,7 +44,7 @@ func (d *dcount) ConsumeAsPartial(partial vector.Any) {
 	if partial.Type() != super.TypeBytes || partial.Len() != 1 {
 		panic("dcount: invalid partial")
 	}
-	b, _ := vector.BytesValue(partial, 1)
+	b, _ := vector.BytesValue(partial, 0)
 	var s hyperloglog.Sketch
 	if err := s.UnmarshalBinary(b); err != nil {
 		panic(fmt.Errorf("dcount: unmarshaling partial: %w", err))

